sycl/doc/extensions/Bf16Conversion/SYCL_INTEL_bf16_conversion.asciidoc

```diff
@@ -132,15 +135,16 @@ public:
   bfloat16(const float &a);
   bfloat16 &operator=(const float &a);
 
-  // Convert from bfloat16 to float
+  // Convert bfloat16 to floating-point types
   operator float() const;
+  operator sycl::half() const
 
-  // Get bfloat16 as uint16.
-  operator storage_t() const;
-
-  // Convert to bool type
+  // Convert bfloat16 to bool type
   explicit operator bool();
 
+  // Get bfloat16 as raw bits.
+  storage_t raw() const;
```

Comment:
The names of the functions `from_bits()` and `raw()` seem inconsistent.  The names should be more similar since they are doing corresponding conversions.  For example, they could be named `from_bits()` and `bits()` or `from_raw()` and `raw()`.

Reply:
Looks reasonable to me. I'd prefer `from_bits()` and `bits()` variant. @GhGmc2 what do you think?

Reply:
Is there a reason to add these new functions instead of re-using `bit_cast`?  `bit_cast<uint16_t>(a)` looks to be equivalent to `a.raw()`, and `bit_cast<bfloat16>(a)` equivalent to `from_bits(a)`.

Reply:
Hmm, I can't think of any specific reason and it seems like it really can work with `bit_cast` out of the box: http://cmplrexplorer.intel.com/z/KG4rfq
But it becomes a bit wordy, especially in case when we want to initialize `bfloat16` with a literal
`bfloat16 X = sycl::bit_cast<bfloat16>(0);`
doesn't work, we need to specify the `from` type
`bfloat16 Y = sycl::bit_cast<bfloat16, uint16_t>(0);`

@GhGmc2 any objections if we remove `raw` and `from_bits`?

Reply:
May I know any progress on this discussion?
Looks like it's blocked for about long while.

We are going to add bfloat16 support in PyTorch master code base, if JIRA CMPLRLLVM-36893 is fixed.
But looks like the progress of bfloat16 in DPC++ compiler is waiting for this PR.
Could you help on this? @AlexeySotkin  @Pennycook  @gmlueck 

For from_bits, looks like PyTorch BFloat16 class has same API.
For completeness of sycl::bfloat16, perhaps we can reserve it?
Or, we can make it as a wrapper of sycl::bit_cast<bfloat16, uint16_t>?

Reply:
It seems like the PyTorch BFloat16 could very easily use `sycl::bit_cast`, so I don't think that one use case should drive the decision.  I think the main argument is whether `sycl::bit_cast<bfloat16, uint16_t>(0)` is too verbose.  I don't know how common it will be to create a `bfloat16` from a uint16 literal, so I don't have a strong opinion.  @Pennycook?

Reply:
I don't feel strongly about this either, but if the concern is specific to creating from literals, a user-defined literal might be a good solution there.  Then users would have options depending on what they're trying to do:
```c++
// Convert x to bit-wise representation
uint16_t bits = sycl::bit_cast<uint16_t>(x);

// Convert bits back to a bfloat16
bfloat16 y = sycl::bit_cast<bfloat16>(bits);

// Create bfloat16 directly from a literal
bfloat16 z = 1.234_bf16;
```

Reply:
Then, our conclusion is we will have no from_bits in bfloat16 class.
For bit_cast, did we agree on user-defined solution?
@Pennycook  @gmlueck  @rdeodhar 

// Convert x to bit-wise representation
uint16_t bits = sycl::bit_cast<uint16_t>(x);

// Convert bits back to a bfloat16
bfloat16 y = sycl::bit_cast<bfloat16>(bits);

// Create bfloat16 directly from a literal
bfloat16 z = 1.234_bf16;

Reply:
Any conclusion?
@Pennycook @gmlueck @rdeodhar

Reply:
I think adding these two function is not strictly necessary so we should not add them.

Reply:
Makes sense.  If we decide later that we want them, it is easy to add things to the extension.

Reply:
Based on the earlier discussion we will not be proceeding with this change. So this PR can be closed, right?

Reply:
No, I think there are still some parts of this PR that we want:

* Removes the  implicit conversion to `uint16_t`.
* Adds an implicit conversion to `sycl::half`.